pkg/querier/queryrange/roundtrip.go

```diff
@@ -29,17 +30,17 @@ import (
 
 // Config for query_range middleware chain.
 type Config struct {
-	SplitQueriesByDay    bool `yaml:"split_queries_by_day"`
-	AlignQueriesWithStep bool `yaml:"align_queries_with_step"`
-	ResultsCacheConfig   `yaml:"results_cache"`
-	CacheResults         bool `yaml:"cache_results"`
-	MaxRetries           int  `yaml:"max_retries"`
+	SplitQueriesByInterval time.Duration `yaml:"split_queries_by_interval"`
+	AlignQueriesWithStep   bool          `yaml:"align_queries_with_step"`
+	ResultsCacheConfig     `yaml:"results_cache"`
+	CacheResults           bool `yaml:"cache_results"`
+	MaxRetries             int  `yaml:"max_retries"`
 }
 
 // RegisterFlags adds the flags required to config this to the given FlagSet.
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.MaxRetries, "querier.max-retries-per-request", 5, "Maximum number of retries for a single request; beyond this, the downstream error is returned.")
-	f.BoolVar(&cfg.SplitQueriesByDay, "querier.split-queries-by-day", false, "Split queries by day and execute in parallel.")
+	f.DurationVar(&cfg.SplitQueriesByInterval, "querier.split-queries-by-interval", 0, "Split queries by an interval and execute in parallel.")
```

Comment:
I would also add `0 disables it.` to the flag description.

Reply:
hum does that means we should may be revise the loki bucketing scheme ?

Reply:
I am not sure tbh, so by splitting cleanly across bucketing scheme we ensured that no two queriers downloaded and acted on the same chunks.